### PR TITLE
[LoongArch64] Remove the redundant check of DivideByZeroException for GT_UDIV/GT_UMOD in CodeGen::genCodeForDivMod().

### PR DESCRIPTION
**Pull request by LuckyXu-HF**
This PR is to optimize the duplicate `DivideByZeroException` check for `GT_UDIV/GT_UMOD` in `CodeGen::genCodeForDivMod()`.
This situation is mentioned in PR https://github.com/dotnet/runtime/pull/98648 for RISC-V.

### CODE REVIEW

**Comment by LuckyXu-HF** [comment]
Hi @clamp03 ,thank you very much for your reminder.
Maybe RISC-V can also remove it.

**Comment by clamp03** [comment]
@LuckyXu-HF Thank you.
@tomeksowi Could you remove it in RISC-V too?

**Comment by LuckyXu-HF** [comment]
@shushanhf  @jakobbotsch 
Could you please review this PR? Thanks.

**Review by jakobbotsch — APPROVED** [review]

File: src/coreclr/jit/codegenloongarch64.cpp [hunk 1]
```diff
@@ -2514,70 +2514,28 @@ void CodeGen::genCodeForDivMod(GenTreeOp* tree)
...
-                }
-                else
-                {
-                    ins = INS_mod_wu;
-                }
+                ins = tree->OperIs(GT_UDIV) ? INS_div_wu : INS_mod_wu;
 
                 // TODO-LOONGARCH64: here is just for signed-extension ?
                 emit->emitIns_R_R_I(INS_slli_w, EA_4BYTE, Reg1, Reg1, 0);
                 emit->emitIns_R_R_I(INS_slli_w, EA_4BYTE, divisorReg, divisorReg, 0);
```
Comment (tomeksowi) [inline src/coreclr/jit/codegenloongarch64.cpp:2534]: [Unrelated to this change] Speaking of optimizations, I removed the argument sign-extensions in RISC-V and I believe you can too if (div|mod)w on LoongArch have the same semantics.

**Comment by tomeksowi** [comment]
> @tomeksowi Could you remove it in RISC-V too?

Done in #99228. Thank you all.

**Review by tomeksowi — APPROVED** [review]

File: src/coreclr/jit/codegenloongarch64.cpp (hunk 1 above)
Comment (shushanhf) [reply src/coreclr/jit/codegenloongarch64.cpp:2534]: Maybe I think it is better to keep this sined-extension conservatively as the optimization is too complicated.
The best is to do it before entering the Codegen.

Comment (LuckyXu-HF) [reply src/coreclr/jit/codegenloongarch64.cpp:2534]: Yes, the sign-extensions here are necessary temporarily now.
We will systemic optimize here in the future.

**Comment by jakobbotsch** [comment]
@shushanhf Can you approve as well? I'll merge it once you've reviewed.

**Review by shushanhf — APPROVED** [review]
I'm sorry to forget adding Approved.
LGTM
Thanks

**Pull request merged**